octodns/provider/ns1.py

```diff
@@ -994,9 +1055,25 @@ def _params_for_dynamic_A(self, record):
             if us_state:
                 meta['us_state'] = sorted(us_state)
 
-            regions[pool_name] = {
-                'meta': meta,
-            }
+            if not georegion and not country and not us_state:
+                # This is the catchall pool. Modify the pool name in the record
+                # being pushed
+                # NS1 regions are indexed by pool names. Any reuse of pool
+                # names in the rules will result in overwriting of the pool.
+                # Reuse of pools is in general disallowed but for the case of
+                # the catchall pool - to allow legitimate usecases.
+                # The pool name renaming is done to accommodate for such a
+                # reuse.
+                # (We expect only one catchall per record. Any associated
+                # validation is expected to covered under record validation)
+                catchall_pool_name = pool_name
+                regions['{}{}'.format(self.CATCHALL_PREFIX, pool_name)] = {
+                    'meta': meta,
+                }
+            else:
+                regions[pool_name] = {
+                    'meta': meta,
+                }
```

Comment:
Rather than repeating the regions addition, it can come out of the if/else and the pool_name can just be updated.

```suggestion
                pool_name = '{}{}'.format(self.CATCHALL_PREFIX, pool_name)

            regions[pool_name] = {
                'meta': meta,
            }
```